src/renderer/components/dock/edit-resource/view.tsx

```diff
@@ -4,169 +4,82 @@
-          }
-        },
-      ),
-      autorun(() => {
-        const store = this.store;
-        const tabData = this.tabData;
-        const resource = this.resource;
-
-        if (!resource && store && tabData) {
-          if (store.isLoaded) {
-            // auto-close tab when resource removed from store
-            this.props.closeTab(this.props.tab.id);
-          } else if (!store.isLoading) {
-            // preload resource for editing
-            store.loadFromPath(tabData.resource).catch(noop);
-          }
-        }
-      }),
-    ]);
-  }
-
-  get tabId() {
-    return this.props.tab.id;
-  }
-
-  get store() {
-    return this.props.editResourceStore.getStore(this.props.tab.id);
-  }
-
-  get resource() {
-    return this.props.editResourceStore.getResource(this.tabId);
-  }
-
-  get tabData() {
-    return this.props.editResourceStore.getData(this.tabId);
-  }
-
-  async save({ resource, store, tabData }: SaveDraftArgs) {
-    if (this.error) {
-      return null;
-    }
-
-    const currentVersion = yaml.load(this.draft);
-    const firstVersion = yaml.load(tabData.firstDraft ?? this.draft);
-    const patches = createPatch(firstVersion, currentVersion);
-    const updatedResource = await store.patch(resource, patches);
-
-    this.props.editResourceStore.clearInitialDraft(this.tabId);
-
-    return (
-      <p>
-        {updatedResource.kind}
-        {" "}
-        <b>{updatedResource.getName()}</b>
-        {" updated."}
-      </p>
-    );
-  }
-
-  render() {
-    const { tabId, error, draft, tabData, resource, store } = this;
-
-    if (!tabData || !resource || !store) {
-      return <Spinner center />;
-    }
-
+const NonInjectedEditResource = observer(
+  ({ model, tab: { id: tabId }}: EditResourceProps & Dependencies) => {
     return (
       <div className="EditResource flex column">
-        <InfoPanel
-          tabId={tabId}
-          error={error}
-          submit={() => this.save({ resource, store, tabData })}
-          submitLabel="Save"
-          submittingMessage="Applying.."
-          controls={(
-            <div className="resource-info flex gaps align-center">
-              <span>Kind:</span>
-              <Badge label={resource.kind} />
-              <span>Name:</span>
-              <Badge label={resource.getName()} />
-              <span>Namespace:</span>
-              <Badge label={resource.getNs() || "global"} />
-            </div>
-          )}
-        />
-        <EditorPanel
-          tabId={tabId}
-          value={draft}
-          onChange={draft => {
-            this.error = "";
-            this.draft = tabData.draft = draft;
-          }}
-          onError={error => this.error = String(error)}
-        />
+        {model.shouldShowErrorAboutNoResource && (
+          <Notice>
+            Resource not found
+          </Notice>
+        )}
+
+        {!model.shouldShowErrorAboutNoResource && (
```

Comment:
nit: could combine these

Reply:
I'd say this is easier to read because there is less depth.